frontend/src/components/UpstreamProvider/LinkUpstreamProvider.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 New Vector Ltd.
+// Copyright 2023, 2024 The Matrix.org Foundation C.I.C.
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+// Please see LICENSE in the repository root for full details.
+
+import IconUserAdd from "@vector-im/compound-design-tokens/assets/web/icons/user-add";
+import { Button } from "@vector-im/compound-web";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import { FragmentType, graphql, useFragment } from "../../gql";
+import LoadingSpinner from "../LoadingSpinner";
+
+const FRAGMENT = graphql(/* GraphQL */ `
+  fragment LinkUpstreamProvider_provider on UpstreamOAuth2Provider {
+    id
+    humanName
+    upstreamOauth2LinksForUser {
+      id
+      provider {
+        id
+      }
```

Comment:
Why are we selecting this?

Reply:
Good question. I have no idea what I wanted to do here anymore. Lets get rid of it :D